P3120 Козаева Яна Александровна task-5

yanakozaeva opened the pull request.

yanakozaeva committed:
,

thejerome commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/IFMO_CPP_programming_20190915/IFMO_CPP_programming_20190915/pullRequest?prid=4532058&bid=15254865)

thejerome commented on src/main/cpp/t07_swap.cpp:
```diff
@@ -19,5 +19,23 @@
+	for (int i = 0; i < n; i++) {
+		cin >> mas[i];
+	}
+	int numb = 0;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'numb' can be reduced.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'numb' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'numb' can be reduced.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'numb' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'numb' can be reduced.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'numb' is assigned a value that is never used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome commented on src/main/cpp/t08_cycleswap.cpp:
```diff
@@ -19,5 +19,25 @@
+	int c = 0;
+	c = mas[0];
+	mas[0] = mas[n - 1];
+	int b = 0;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'b' can be reduced.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'b' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'b' can be reduced.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'b' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'b' can be reduced.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'b' is assigned a value that is never used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome commented on src/main/cpp/t05_minodd.cpp:
```diff
@@ -28,5 +28,42 @@
+
+
+
+	if (s = 1)
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is reassigned a value before the old one has been used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is assigned a value that is never used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is reassigned a value before the old one has been used.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

yanakozaeva committed:
Done

thejerome commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/IFMO_CPP_programming_20190915/IFMO_CPP_programming_20190915/pullRequest?prid=4532058&bid=15254865)

yanakozaeva committed:
Changes to be committed:
	modified:   src/main/cpp/t05_minodd.cpp
	modified:   src/main/cpp/t10_unique.cpp

thejerome commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058&bid=15254865)

thejerome commented on src/main/cpp/t05_minodd.cpp:
```diff
@@ -28,5 +28,42 @@
+
+
+
+	if (s = 1)
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is assigned a value that is never used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 's' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

yanakozaeva committed:
Changes to be committed:
	modified:   src/main/cpp/t05_minodd.cpp
	modified:   src/main/cpp/t07_swap.cpp
	modified:   src/main/cpp/t08_cycleswap.cpp

thejerome commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058&bid=15254865)

thejerome commented on src/main/cpp/t08_cycleswap.cpp:
```diff
@@ -19,5 +19,27 @@
+	int c = 0;
+	c = mas[0];
+	mas[0] = mas[n - 1];
+	int b = 0;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'b' is assigned a value that is never used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

thejerome commented on src/main/cpp/t05_minodd.cpp:
```diff
@@ -28,5 +28,42 @@
+		if (mas[i] < min && mas[i] % 2 == 1)
+		{
+			min = mas[i];
+			s ++;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Incrementing a variable of type 'bool' with postfix operator++ is deprecated by the C++ Standard. You should assign it the value 'true' instead.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

yanakozaeva committed:
Changes to be committed:
	modified:   src/main/cpp/t05_minodd.cpp
	modified:   src/main/cpp/t08_cycleswap.cpp

thejerome commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058&bid=15254865)

thejerome commented on src/main/cpp/t05_minodd.cpp:
```diff
@@ -29,4 +29,45 @@ using namespace std;
+		if ((mas[i] < min) && ((mas[i] % 2 == 1) || (mas[i] % 2 == -1)))
+		{
+			min = mas[i];
+			s++;
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Incrementing a variable of type 'bool' with postfix operator++ is deprecated by the C++ Standard. You should assign it the value 'true' instead.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4532058)

yanakozaeva committed:
Changes to be committed:
	modified:   src/main/cpp/t05_minodd.cpp

thejerome reviewed (commented):
Score: **90**
